### PR TITLE
StudentSemesterCourse get all service task

### PR DESCRIPTION
**Pull request by levirgon**
# Task complete
# All tests implemented.

### CODE REVIEW

File: OtripleS.Web.Api/Services/StudentSemesterCourses/StudentSemesterCourseService.Exceptions.cs [hunk 1]
```diff
@@ -9,13 +9,35 @@
...
+                throw CreateAndLogCriticalDependencyException(sqlException);
+            }
+            catch (DbUpdateException dbUpdateException)
+            {
+                throw CreateAndLogDependencyException(dbUpdateException);
+            }catch (Exception exception)
+            {
+                throw CreateAndLogServiceException(exception);
+            }
+        }
```
Comment (tobidub) [inline OtripleS.Web.Api/Services/StudentSemesterCourses/StudentSemesterCourseService.Exceptions.cs:40]: I think there is one new line missing

File: OtripleS.Web.Api.Tests.Unit/Services/StudentSemesterCourseServiceTests/StudentSemesterCourseServiceTests.Exceptions.Retrieve.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        }
+
+        [Fact]
+        public void ShouldThrowDependencyExceptionOnRetrieveAllWhenDbExceptionOccursAndLogIt()
+        {
+            // given
+            var databaseUpdateException = new DbUpdateException();
+
+            var expectedSemesterCourseDependencyException =
+                new SemesterCourseDependencyException(databaseUpdateException);
```
Comment (tobidub) [inline OtripleS.Web.Api.Tests.Unit/Services/StudentSemesterCourseServiceTests/StudentSemesterCourseServiceTests.Exceptions.Retrieve.cs:53]: Shouldnt that be a StudentSemesterCourseDependencyException?

Comment (devmanzur) [reply OtripleS.Web.Api.Tests.Unit/Services/StudentSemesterCourseServiceTests/StudentSemesterCourseServiceTests.Exceptions.Retrieve.cs:53]: Yes, Sorry, I had so many typing mistakes regarding this two entities. I will fix this

File: OtripleS.Web.Api.Tests.Unit/Services/StudentSemesterCourseServiceTests/StudentSemesterCourseServiceTests.Logic.cs [hunk 3]
```diff
@@ -30,12 +31,12 @@ public async Task ShouldCreateStudentSemesterCourseAsync()
             StudentSemesterCourse expectedStudentSemesterCourse = storageStudentSemesterCourse;
 
             this.dateTimeBrokerMock.Setup(broker =>
-                broker.GetCurrentDateTime())
-                    .Returns(dateTime);
+                    broker.GetCurrentDateTime())
+                .Returns(dateTime);
```
Comment (hassanhabib) [inline OtripleS.Web.Api.Tests.Unit/Services/StudentSemesterCourseServiceTests/StudentSemesterCourseServiceTests.Logic.cs:35]: Push this two tabs further

Comment (devmanzur) [reply OtripleS.Web.Api.Tests.Unit/Services/StudentSemesterCourseServiceTests/StudentSemesterCourseServiceTests.Logic.cs:34]: Fixed

**Comment by hassanhabib** [comment]
@levirgon  @devmanzur please address conflicts

**Pull request merged**